### PR TITLE
first name, last name changes, protractor, karma

### PR DESCRIPTION
**Pull request by pooja-koul-incontact**

### CODE REVIEW

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    userAssigned: ''
+  };
+  ngOnInit() {
+  }
+  agInit(params) {
+    this.rowData = params.data;
+  }
+  getStatus() {
+    return this.rowData.status === 'Mapped';
+  }
```
Comment (KMayuri25) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.ts]: can we put this in AppConstants

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.html [hunk 2]
```diff
@@ -17,7 +17,7 @@ <h4 class="modal-title module-title add-user-title" id="myModalLabel">Update Rul
                   <div class="error-message" *ngIf="control.ruleName.errors.required && updateRuleForm.controls['ruleName'].touched">This field is required.</div>
                   <div class="error-message" *ngIf="control.ruleName.errors.pattern">Input is not alphanumeric.</div>
                 </div>
-                <div class="error-message" *ngIf="ruleNameError">Rule Name already exists.</div>
+                <div class="error-message" id="ruleNameError" *ngIf="ruleNameError">Rule Name already exists.</div>
```
Comment (KMayuri25) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.html]: same

File: e2e/src/RuleMapping/RuleMapping.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  let buPage: BuPage;
+  const testUtils: TestUtils = new TestUtils();
+  beforeAll(() => {
+    browser.wait(EC.browser.driver.sleep(5000));
+  });
+  afterAll(() => {
+    testUtils.presencesynclogout();
+  });
+  beforeEach(() => {
+    buPage = new BuPage();
```
Comment (Susmith-roy-incontact) [inline e2e/src/RuleMapping/RuleMapping.e2e-spec.ts]: move it to beforeAll()

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <div class="status-icon-display" *ngIf="getStatus()">
+    <span class="completed-icon-to-display"></span>
+  </div>
+  <div class="status-icon-display" *ngIf="!getStatus()">
+    <span class="failed-icon-to-display"></span>
+  </div>&nbsp;
+  {{(rowData.status === 'Mapped') ? 'Mapped' : 'Not Mapped'}}
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.html]: Use it from AppConstatnts

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+@Component({
+  selector: 'app-bu-rule-mapping-user-status',
+  templateUrl: './bu-rule-mapping-user-status.component.html',
+  styleUrls: ['./bu-rule-mapping-user-status.component.css']
+})
+export class BuRuleMappingUserStatusComponent implements OnInit {
+
+  constructor() { }
+  rowData = {
+    ruleName: '',
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-user-status/bu-rule-mapping-user-status.component.ts]: Do we need the whole object instaed can we use one object

**Pull request merged**